main/src/library/Fixpoint/Solver.flix

```diff
@@ -61,15 +61,11 @@ namespace Fixpoint {
         case Datalog(facts, rules) => Datalog(restrict(facts, Set#{p}), rules)
     }
 
-    pub def foldRight(d: Datalog[v], p: PredSym, i: b, f: (Array[v], b) -> b): b =
-        let Datalog(facts, _) = project(p, d);
-        Array.foldRight(factToTuple >> f, i, facts) as & Pure
-
     ///
     /// Returns all facts in `d` associated with the predicate symbol `p`.
     ///
     pub def facts0(p: PredSym, d: Datalog[Boxed]): Array[v] with Boxable[v] =
-        ??? // TODO: Benjamin
+        (factsOf(p, d) |> Array.map(_ -> () as v)) as & Pure
```

Comment:
This cast to `v` makes me a bit uneasy.

Reply:
Did you test it?

Reply:
Yeah your testcase prints `() :: Nil` and
```
def main(_: Array[String]): Int32 & Impure =
    let s = facts D #{ A(1, 2, 3). B("hello"). A(4, 5, 6). C(42). K(1, 2, 3, 4, 5). D() :- if false. } |> identity;
    s |> Array.toList |> println;
    0
```
prints `Nil`. So it seems to work.